File: kraken/src/main/java/glitch/kraken/services/IngestService.java
package glitch.kraken.services;

import glitch.api.http.Routes;
import glitch.api.objects.json.interfaces.OrdinalList;
import glitch.kraken.GlitchKraken;
import glitch.kraken.object.json.Ingest;
import glitch.kraken.object.json.collections.Ingests;
import glitch.service.AbstractHttpService;
import reactor.core.publisher.Flux;

public class IngestService extends AbstractHttpService {
    public IngestService(GlitchKraken rest) {
        super(rest.getClient(), rest.getHttpClient());
    }

    public Flux<Ingest> getIngestServerList() {
        return exchangeTo(Routes.get("/ingests").newRequest(), Ingests.class)
                .flatMapIterable(OrdinalList::getData);
    }
}

File: kraken/src/main/kotlin/io/glitchlib/v5/service/IngestService.kt
package io.glitchlib.v5.service

import io.glitchlib.GlitchClient
import io.glitchlib.internal.http.bodyFlowable
import io.glitchlib.model.OrdinalList
import io.glitchlib.v5.internal.AbstractKrakenService
import io.glitchlib.v5.model.json.Ingest

class IngestService(client: GlitchClient) : AbstractKrakenService(client) {
    val ingestServerList
        get() = get<OrdinalList<Ingest>>("/ingests").bodyFlowable
}
